src/main/objects.c

```diff
@@ -991,16 +991,29 @@ static SEXP inherits3(SEXP x, SEXP what, SEXP which)
     else
 	PROTECT(klass = R_data_class(x, FALSE));
 
+    int nprot = 1;
+
+    if (inherits(what, "R7_class"))
```

Comment:
We might want to make the `isString()` test come first since that will be the common case and should be faster than looking at inheritance.

Reply:
fixed in https://github.com/r-devel/r-svn/pull/100/commits/307253e698199dbc6b334c9fd76534e481f786e7